File: arkts_static_test/03_Types/ets_cases/3.16_Array_Types/cross_lang_verify/TYP_03_16_Array_Types.swift
/**
 * Swift cross-language verification for 3.16 Array Types
 */
import Foundation

print("=== 3.16 Array Types ===")

// 1. Swift 数组声明
let arr1: [Int] = [1, 2, 3]
let arr2: [String] = ["a", "b", "c"]
print("Swift: [Int] count=\(arr1.count)")
print("Swift: [String] count=\(arr2.count)")

// 2. Swift 数组是值类型（struct）
var arr3 = arr1
arr3.append(4)
print("Swift: arr1.count=\(arr1.count) arr3.count=\(arr3.count)")

// 3. Swift 数组可迭代
var sum = 0
for n in arr1 {
    sum += n
}
print("Swift: sum=\(sum)")

// 4. Swift 数组索引访问
print("Swift: arr1[0]=\(arr1[0]) arr1[1]=\(arr1[1])")

// 5. Swift 数组可变性
var mutableArr = [1, 2, 3]
mutableArr.append(4)
mutableArr.remove(at: 0)
print("Swift: mutableArr=\(mutableArr)")

// 6. Swift Array 与 NSArray
let nsArray: NSArray = [1, 2, 3] as NSArray
print("Swift: NSArray bridge=\(nsArray)")

File: arkts_static_test/03_Types/ets_cases/3.16_Array_Types/cross_lang_verify/TYP_03_16_Array_Types.java
/**
 * Java cross-language verification for 3.16 Array Types
 */
class TYP_03_16_Array_Types {
    public static void main(String[] args) {
        // 1. Java 数组声明
        int[] arr1 = {1, 2, 3};
        String[] arr2 = {"a", "b", "c"};
        System.out.println("3.16: int[] length=" + arr1.length);
        System.out.println("3.16: String[] length=" + arr2.length);

        // 2. Java 数组是 Object 子类型
        Object o = arr1;
        System.out.println("3.16: array instanceof Object: " + (o instanceof Object));

        // 3. Java 数组可迭代（使用 for-each）
        int sum = 0;
        for (int n : arr1) {
            sum += n;
        }
        System.out.println("3.16: sum=" + sum);

        // 4. Java 数组索引访问
        System.out.println("3.16: arr1[0]=" + arr1[0] + " arr1[1]=" + arr1[1]);

        // 5. Java 没有 FixedArray 概念
        System.out.println("3.16: Java has no FixedArray concept");

        // 6. Java 数组长度固定
        System.out.println("3.16: Java array length is fixed after creation");
    }
}
